geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/strategies/AttributeFilterStrategy.scala

```diff
@@ -62,7 +62,7 @@ trait AttributeFilterStrategy[DS <: GeoMesaDataStore[DS, F, W], F <: WrappedFeat
         val cost = stats.flatMap(_.getCount(sft, f, exact = false)).getOrElse {
           val binding = sft.getDescriptor(attribute).getType.getBinding
           val bounds = FilterHelper.extractAttributeBounds(f, attribute, binding)
-          if (bounds.forall(_.isExact)) {
+          if (bounds.precise && bounds.forall(_.isExact)) {
```

Comment:
For context, what's the meaning of precise and exact here?

I suppose I'm curious about how something aould be exact and not precise.

Reply:
It is a bit confusing... i did doc it in the scala docs though :)
precise means that the ranges exactly match the filter, and so there won't be any false positives returned. isExact means that it's a single equality check, vs a range.
got any suggestions for better names?

Reply:
Ok, so precise about whether or not we are querying with 'slop', and exact is about 'Gets' versus 'Scans'?

Naming things is hard.  I was mainly curious.  We can leave it as is...

Are there non-precise 'Gets'?

Reply:
i don't think there are non-precise gets, although technically it is allowed... i was originally going to call exact 'isEquals' or something but that seemed confusing too...